src/displayapp/screens/Dice.cpp

```diff
@@ -0,0 +1,231 @@
+  lv_obj_set_size(numDiceDecBtn, 70, 50);
+  lv_obj_set_pos(numDiceDecBtn, 5, 105);
+  lv_obj_set_ext_click_area(numDiceDecBtn, 5, 5, 10, 5);
+  numDiceDecBtnLabel = lv_label_create(numDiceDecBtn, nullptr);
+  lv_label_set_text(numDiceDecBtnLabel, "-");
+
+  diceSizeIncBtn = lv_btn_create(configCont, nullptr);
+  diceSizeIncBtn->user_data = this;
+  lv_obj_set_event_cb(diceSizeIncBtn, dice_size_inc_event_handler);
+  lv_obj_set_size(diceSizeIncBtn, 70, 50);
+  lv_obj_set_pos(diceSizeIncBtn, 85, 5);
+  lv_obj_set_ext_click_area(diceSizeIncBtn, 5, 5, 5, 10);
+  diceSizeIncBtnLabel = lv_label_create(diceSizeIncBtn, nullptr);
+  lv_label_set_text(diceSizeIncBtnLabel, "+");
+
+  diceSizeLabelCont = lv_cont_create(configCont, nullptr);
+  lv_obj_set_size(diceSizeLabelCont, 80, 40);
+  lv_obj_set_pos(diceSizeLabelCont, 80, 60);
+  lv_obj_set_style_local_bg_color(diceSizeLabelCont, LV_CONT_PART_MAIN, LV_STATE_DEFAULT, LV_COLOR_BLACK);
+  lv_cont_set_layout(diceSizeLabelCont, LV_LAYOUT_CENTER);
+  diceSizeLabel = lv_label_create(diceSizeLabelCont, nullptr);
+  lv_label_set_text_fmt(diceSizeLabel, "d%d", 6);
+
+  diceSizeDecBtn = lv_btn_create(configCont, nullptr);
+  diceSizeDecBtn->user_data = this;
+  lv_obj_set_event_cb(diceSizeDecBtn, dice_size_dec_event_handler);
+  lv_obj_set_size(diceSizeDecBtn, 70, 50);
+  lv_obj_set_pos(diceSizeDecBtn, 85, 105);
+  lv_obj_set_ext_click_area(diceSizeDecBtn, 5, 5, 10, 5);
+  diceSizeDecBtnLabel = lv_label_create(diceSizeDecBtn, nullptr);
+  lv_label_set_text(diceSizeDecBtnLabel, "-");
+}
+
+void Dice::rollDice() {
+  uint8_t roll = 0;
+  uint16_t result = 0;
+
+  lv_label_set_text(resultTotalLabel, "");
+  lv_label_set_text(resultIndRollsLabel, "");
+
+  for (int i = 0; i < numDice; i++) {
+    roll = rand() % diceSizes[diceSizeIndex] + 1;
+    result += roll;
+
+    if (numDice > 1) {
+      if (i == 0) {
+        lv_label_ins_text(resultIndRollsLabel, 0, "(");
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, std::to_string(roll).c_str());
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, ",");
+      } else if (i < numDice - 1) {
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, std::to_string(roll).c_str());
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, ",");
+      } else {
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, std::to_string(roll).c_str());
+        lv_label_ins_text(resultIndRollsLabel, LV_LABEL_POS_LAST, ")");
+      }
+    }
+  }
+
+  lv_label_set_text_fmt(resultTotalLabel, "%d", result);
+}
+
+Dice::Dice(DisplayApp* app,
+           Controllers::MotionController& motionController,
+           Controllers::MotorController& motorController,
+           System::SystemTask& systemTask)
+  : Screen(app),
+    systemTask {systemTask},
+    motionController {motionController},
+    motorController {motorController},
+    numDice(1),
+    diceSizeIndex(3) {
+
+  app->SetTouchMode(DisplayApp::TouchModes::Polling);
```

Comment:
This improves responsiveness, but is kind of hacky. I think it's fine though.

Reply:
I saw that this was something that other apps had done so I included it as well.  If it's fine as-is I will just leave it.